x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterTests.java

```diff
@@ -328,4 +342,15 @@ private RemoteAccessServerTransportFilter getNodeRemoteAccessFilter(Set<String>
             new SecurityContext(settings, threadContext)
         );
     }
+
+    private static Authentication replaceWithInternalUserAuthcForHandshake(String action, Authentication authentication) {
+        return Set.of(TransportService.HANDSHAKE_ACTION_NAME, TransportActionProxy.getProxyAction(TransportService.HANDSHAKE_ACTION_NAME))
```

Comment:
Off the top of my head, I don't think the handshake action is ever proxied, so we should tweak the allowlist but I think it makes sense to make this fix separately from addressing the general test failure.